NuKeeper/Engine/RepositoryUpdater.cs

```diff
@@ -106,7 +106,7 @@ public class RepositoryUpdater : IRepositoryUpdater
 
         private static void GitInit(IGitDriver git, RepositoryData repository)
         {
-            git.Clone(repository.Pull.Uri);
+            git.Clone(repository.Pull.Uri, repository.DefaultBranch);
```

Comment:
Can't we just check here if `targetBranch is set then do a gitclone? Think you can just check if `repository.DefaultBranch !=  git.GetCurrentHead()`

Reply:
repository.DefaultBranch != git.GetCurrentHead() only works áfter the clone
So then you are back to the extra checkout.
LibGit2Sharp supports the clone-with-branch out of the box

Reply:
Awesome i didn't know that